mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -189,11 +232,21 @@ class ChiSqSelector @Since("1.3.0") (
    */
   @Since("1.3.0")
   def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
-    val indices = Statistics.chiSqTest(data)
-      .zipWithIndex.sortBy { case (res, _) => -res.statistic }
-      .take(numTopFeatures)
-      .map { case (_, indices) => indices }
-      .sorted
+    val chiSqTestResult = Statistics.chiSqTest(data)
+    val features = selectorType match {
+      case ChiSqSelectorType.KBest => chiSqTestResult
+        .zipWithIndex.sortBy { case (res, _) => -res.statistic }
```

Comment:
I think it would still be better to sort here since it's actually simpler and not expensive. Do you feel strongly about it?

Otherwise, I have just one last thing I noted when reading all the code again -- how about changing "Fpr" to "FPR" because it does stand for "False Positive Rate"?